internal/mode/static/state/dataplane/types.go

```diff
@@ -131,6 +133,28 @@ type HTTPRequestRedirectFilter struct {
 	StatusCode *int
 }
 
+// HTTPURLRewriteFilter rewrites HTTP requests.
+type HTTPURLRewriteFilter struct {
+	// Hostname is the hostname of the rewrite.
+	Hostname *string
+	// Path is the path of the rewrite.
+	Path *HTTPPathModifier
+}
+
+// HTTPPathModifier defines configuration for path modifiers.
+type HTTPPathModifier struct {
+	// ReplaceFullPath specifies the value with which to replace the full path of a request during
+	// a rewrite or redirect.
+	ReplaceFullPath string
```

Comment:
Are you able to combine `ReplaceFullPath` and `ReplacePrefixMatch` with a single string field? I'm not sure if I'm missing something, but if you already have separate booleans for `FullPath` and `PrefixMatch` and this Modifier is only for one or the other, does the string field need to be split up?

Reply:
Yeah that's a good point, I can probably combine.